iota/api.py

```diff
@@ -82,6 +82,7 @@ def __init__(self, adapter, testnet=False):
             adapter = resolve_adapter(adapter)
 
         self.adapter = adapter  # type: BaseAdapter
+        self.adapter.set_local_pow(local_pow)
```

Comment:
❔ Putting this in the adapter might be overkill, since it's not really used by the adapter itself; maybe instead do something similar to what we do for `min_weight_magnitude`?

```py
class StrictIota:
    def __init__(self, adapter, testnet=False, local_pow=False):
        self.local_pow = local_pow

    def attach_to_tangle(self, ..., local_pow=None):
        if local_pow is None:
            local_pow = self.local_pow

        return core.AttachToTangleCommand(self.adapter)(
            ....
            local_pow=local_pow,
        )
```

Reply:
It might be an overkill, but greatly simplifies the handling. The problem is, if `local_pow` is associated with the api class, we have to modify every call to `*Command` that somehow uses `AttachToTangleCommand` internally. These commands are in the extended api, like `send_transfer` and `send_trytes`.
On the other hand, if `local_pow` is associated with the adapter (value still can be passed during api instantiation for simplicity and user friendliness) then all we have to do is modify the `AttachToTangleCommand` only, and it handles all direct or indirect calls correctly. The reason is that within the `execute_` function of a `*Command`, we only have access to the adapter, but not the api class.

Reply:
Makes sense to me.  Good thinking!